scripts/lintChanged.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Checks for changes between the common ancestor of origin/main and the
+# current HEAD. 
+
+source scripts/shellUtils.sh
+
+MERGE_BASE_SHA_HASH=$(git merge-base origin/main HEAD)
+# Check if output is empty.
+if [ -z "${MERGE_BASE_SHA_HASH}" ]; then
+    error "git merge-base did not return any output"
+    exit 1
+fi
+
+# Check if the output of merge-base is a single SHA-1 commit hash.
+if ! [[ "${MERGE_BASE_SHA_HASH}" =~ ^[a-fA-F0-9]{40}$ ]]; then
+    error "git merge-base returned unexpected output"
+    exit 1
+fi
+
+DIFF_OUTPUT=$(git --no-pager diff --diff-filter=AMR --name-only "${MERGE_BASE_SHA_HASH}" HEAD -- '*.ts' '*.tsx')
+
+# Check if there were any changes.
+if [ -z "${MERGE_BASE_SHA_HASH}" ]; then
+    error "git diff returned no changed files"
+    exit 1
+fi
+
+# Ensure this script is running inside an npm run context
+if [[ -n "$npm_lifecycle_event" ]]; then
+    eslint --max-warnings=0 --config ./.eslintrc.changed.js "${DIFF_OUTPUT}"
+else
+    error "This script is meant to run by using npm run lint-changed"
+fi
```

Comment:
Is it actually necessary to run this from NPM? If so, then instead of erroring here, then it would be better imo to re-exec the current script in NPM context. You can also refactor to check for `npm_lifecycle_event` to be unset first, since `exec` will replace the currently-running process. Then you can unindent the main case.

```suggestion
if [[ -z "$npm_lifecycle_event" ]]; then
    warn "Re-executing this script from an npm context"
    exec npx -c "NODE_OPTIONS=--max_old_space_size=8192 ${BASH_SOURCE[0]} $*"
fi

eslint --max-warnings=0 --config ./.eslintrc.changed.js "${DIFF_OUTPUT}"
```

Reply:
`eslint` isn't on my path but it is available when I use `npm run`, also I wanted to make sure that we were using the `eslint` for the installed node version not a locally install ed one. But yeah I did consider just making it run in the npm context anyways.
I can make this change.

Reply:
I don't really understand your question, but I think you might be missing what `exec ...` does in bash. Calling `exec <whatever>` _replaces_ the currently-running process with `<whatever>` so anything code after it never gets called. This is solving the exactly problem you are describing of `eslint` not being in your path.

Reply:
I understood what the bash code did, I was more concerned with whether this was a needed change. But either way you're right it did solve the problem. Resolved.